torchlib/find_config_DP.py

```diff
@@ -0,0 +1,193 @@
+        name=f"optuna{cmdln_args.trial_name}",
+    )
+    args.albu_prob = 0.0
+    args.individual_albu_probs = 0.0
+    args.clahe = False
+    args.randomgamma = False
+    args.randombrightness = False
+    args.blur = False
+    args.elastic = False
+    args.optical_distortion = False
+    args.grid_distortion = False
+    args.grid_shuffle = False
+    args.hsv = False
+    args.invert = False
+    args.cutout = False
+    args.shadow = False
+    args.fog = False
+    args.sun_flare = False
+    args.solarize = False
+    args.equalize = False
+    args.grid_dropout = False
+
+    args.differentially_private = True
+    args.target_delta = 1e-5
+    args.noise_multiplier = trial.suggest_float("DP_noise", 0.1, 1.0)
+    args.max_grad_norm = trial.suggest_float("max_grad_norm", 0.1, 2.0)
+
+    if cmdln_args.federated:
+        args.unencrypted_aggregation = cmdln_args.unencrypted_aggregation
+        args.sync_every_n_batch = trial.suggest_int("sigma", 1, 5)
+        args.wait_interval = 0.1
+        args.keep_optim_dict = False
+        if not cmdln_args.unencrypted_aggregation:
+            args.precision_fractional = 16
+        # trial.suggest_categorical(
+        #     "keep_optim_dict", [True, False]
+        # )
+        args.weighted_averaging = trial.suggest_categorical(
+            "weighted_averaging", [True, False]
+        )
+        args.DPSSE = False
+        args.dpsse_eps = 1.0
+        args.microbatch_size = args.batch_size
+    try:
+        best_val_acc, epsilon = main(args, verbose=False, optuna_trial=trial)
+    except Exception as e:
+        print(f"Trial failed with exception {e} and arguments {str(args)}.")
+        return 0
+    if epsilon < 1:
+        warn(f"Epsilon is only {epsilon:.2f}. Seems very low.")
+    return best_val_acc / max(epsilon, 1)
+
+
+if __name__ == "__main__":
+    from argparse import ArgumentParser
+
+    parser = ArgumentParser()
+    parser.add_argument(
+        "--federated", action="store_true", help="Search on federated setting"
+    )
+    parser.add_argument("--data_dir", type=str, help="Path to data")
+    parser.add_argument("--trial_name", type=str, default="", help="Assign identifier")
+    parser.add_argument("--websockets", action="store_true", help="Use websockets")
+    parser.add_argument(
+        "--num_trials", default=40, type=int, help="how many trials to perform"
+    )
+    parser.add_argument(
+        "--visualize", action="store_true", help="Visualize optuna results."
+    )
+    parser.add_argument(
+        "--db_file",
+        type=str,
+        default="sqlite:///model_weights/pneumonia_search_DP.db",
+        help="Database file to store results.",
+    )
+    parser.add_argument(
+        "--unencrypted_aggregation",
+        action="store_true",
+        help="Train model without secure aggregation",
+    )
+    cmdln_args = parser.parse_args()
+    try:
+        study = opt.create_study(
```

Comment:
with a multi-objective problem, why not unleashing the power of Genetic Algorithms!! -> turn it into a `multi_objective` study :)

Reply:
You would not believe what I'm currently running...
![Screenshot 2020-12-31 at 11 13 03](https://user-images.githubusercontent.com/48105485/103405948-39461980-4b59-11eb-8d45-dd890f947609.png)